modules/rup/controllers/estadisticas.ts

```diff
@@ -216,3 +217,174 @@ export async function dashboard(org, prestaciones, desde, hasta) {
+    let matchEstados: any = {};
+
+    /* Condición para filtrar prestaciones que son solicitudes */
+    matchInicial['estados.0.tipo'] = {
+        $in: ['pendiente', 'auditoria']
+    };
+
+    /* Match inicial para filtrar solicitudes
+       según la organización del usuario logueado */
+    const usuarioOrganizacion = user.organizacion;
+    matchSolicitudEntrada['solicitud.organizacion.id'] = new ObjectId(usuarioOrganizacion._id);
+    matchSolicitudSalida['solicitud.organizacionOrigen.id'] = new ObjectId(usuarioOrganizacion._id);
+
+    if (filtros.solicitudDesde && filtros.solicitudHasta) {
+        let fechaCondicion = {
+            $gte: moment(filtros.solicitudDesde).startOf('day').toDate(),
+            $lte: moment(filtros.solicitudHasta).endOf('day').toDate()
+        };
+        matchInicial['solicitud.fecha'] = fechaCondicion;
+    }
+
+    /* Filtro por organización destino */
+    if (filtros.organizaciones) {
+        matchSolicitudEntrada['solicitud.organizacionOrigen.id'] = {
+            $in: filtros.organizaciones.map(org => new ObjectId(org.id))
+        };
+        matchSolicitudSalida['solicitud.organizacion.id'] = {
+            $in: filtros.organizaciones.map(org => new ObjectId(org.id))
+        };
+    }
+
+    if (filtros.profesionalesDestino) {
+        matchFiltros['solicitud.profesional.id'] = {
+            $in: filtros.profesionalesDestino.map(pr => new ObjectId(pr.id))
+        };
+    }
+
+    if (filtros.profesionalesOrigen) {
+        matchFiltros['solicitud.profesionalOrigen.id'] = {
+            $in: filtros.profesionalesOrigen.map(pr => new ObjectId(pr.id))
+        };
+    }
+
+    if (filtros.solicitudesOrigen) {
+        matchFiltros['solicitud.tipoPrestacionOrigen.conceptId'] = {
+            $in: filtros.solicitudesOrigen.map(solic => solic.id)
+        };
+    }
+
+    if (filtros.solicitudesDestino) {
+        matchFiltros['solicitud.tipoPrestacion.conceptId'] = {
+            $in: filtros.solicitudesDestino.map(solic => solic.id)
+        };
+    }
+
+    if (filtros.estados) {
+        matchEstados['ultimoEstado.tipo'] = {
+            $in: filtros.estados
+        };
+    }
+
+    let pipelineEntrada = [
+        /* Filtros */
+        { $match: matchInicial },
+        { $match: matchSolicitudEntrada },
+        { $match: matchFiltros },
+        { $addFields: { ultimoEstado: { $arrayElemAt: ['$estados', -1] }}},
+        { $match: matchEstados}, // Filtro por el último estado
+        { $facet: makeFacet('entrada') }
+    ];
+    let pipelineSalida = [
+        /* Filtros */
+        { $match: matchInicial },
+        { $match: matchSolicitudSalida },
+        { $match: matchFiltros },
+        { $addFields: { ultimoEstado: { $arrayElemAt: ['$estados', -1] }}},
+        { $match: matchEstados}, // Filtro por el último estado
+        { $facet: makeFacet('salida') }
+    ];
+
+    const dataEntrada = toArray(Prestacion.aggregate(pipelineEntrada).cursor({ batchSize: 1000 }).exec());
```

Comment:
```suggestion
    const dataEntrada = Prestacion.aggregate(pipelineEntrada);
```
Desde una nueva versión de mongoose ya no es necesario usar esa función `toArray`